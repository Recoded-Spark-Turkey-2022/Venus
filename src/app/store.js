// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
