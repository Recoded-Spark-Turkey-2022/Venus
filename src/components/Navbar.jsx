import React, { useState } from 'react'
import Button from './Button';
import logo from  '../assets/Logo.svg'

const Navbar = () => {
    let Links =[
      {name:"Home",link:"/"},
      {name:"About",link:"/"},
      {name:"BLog",link:"/"},
      {name:"Contact",link:"/"},
    ];
    let [open,setOpen]=useState(true);
  return (
    <div className='shadow-md w-full fixed top-0 left-0'>
      <div className='md:flex items-center justify-between bg-white py-4 md:px-10 px-7'>
      <div className='font-bold text-2xl cursor-pointer flex items-center font-[Poppins] 
      text-cyan-600'>
        <span className='text-3xl text-cyan-600 mr-1 pt-2'>
        
        <img src={logo}></img>
        </span>
        
      </div>
      
      <div onClick={()=>setOpen(!open)} className='text-3xl absolute right-8 top-6 cursor-pointer md:hidden'>
      <ion-icon  name={open ? 'close':'menu'}></ion-icon>
      </div>

      <ul className={`md:flex md:items-center md:pb-0 pb-12 absolute md:static bg-white md:z-auto z-[-1] left-0 w-full md:w-auto md:pl-0 pl-9 transition-all duration-500 ease-in ${open ? 'top-20 ':'top-[-490px]'}`}>
        {
          Links.map((link)=>(
            <li key={link.name} className='md:ml-8 text-xl md:my-0 my-7'>
              <a href={link.link} className='text-gray-800 hover:text-cyan-600 duration-500'>{link.name}</a>
            </li>
          ))
        }
        <Button>
          Sign in 2
        </Button>
      </ul>
      </div>
    </div>
  )
}

export default Navbar