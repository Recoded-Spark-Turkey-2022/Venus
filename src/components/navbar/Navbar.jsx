import React, { useState } from 'react';
import { useTranslation } from "react-i18next";
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import logo from '../../assets/Logo.svg';
import {
  selectUserLoggedIn,
  setUserLoggedIn,
  setUserLoggedOut,
} from '../../features/userSlice/userSlice';
import { authentication } from '../../firebase/firebase.config';

const Navbar = () => {
  const { t }= useTranslation();
  const isLoggedIn = useSelector(selectUserLoggedIn);
  const dispatch = useDispatch();
  const Links = [
    { name: 'Home', link: '/' },
    { name: 'About', link: '/about' },
    { name: 'Blog', link: '/blogs' },
    { name: 'Contact', link: '/contact' },
  ];

  const [open, setOpen] = useState(true);
  const handleSignOut = async () => {
    try {
      await authentication.signOut();

      dispatch(
        setUserLoggedIn({
          isLoggedIn: false,
        })
      );

      dispatch(setUserLoggedOut());
      localStorage.setItem('token', '');
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <nav className="shadow-md w-full fixed z-1000 top-0 left-0">
      <div className="md:flex items-center justify-between bg-white py-4 md:px-10 px-7">
        <div
          className="font-bold text-2xl cursor-pointer flex items-center font-[Poppins] 
      text-cyan-600"
        >
          <span className="text-3xl text-cyan-600 mr-1 pt-2">
            <img src={logo} alt="logo" />
          </span>
        </div>

        <button
          type="button"
          onClick={() => setOpen(!open)}
          className="text-3xl absolute right-8 top-6 cursor-pointer md:hidden"
        >
          <ion-icon name={open ? 'close' : 'menu'}> </ion-icon>
        </button>

        <ul
          className={`md:flex md:items-center md:pb-0 pb-12 absolute md:static bg-white md:z-auto z-[-1] left-0 w-full md:w-auto md:pl-0 pl-9 transition-all duration-500 ease-in ${
            open ? 'top-30 ' : 'top-[-490px]'
          }`}
        >
          {Links.map((link) => (
            <li key={link.name} className="md:ml-8 text-xl md:my-0 my-7">
              <Link
                to={link.link}
                className="text-cyan-600 hover:text-cyan-600 duration-500 hover:underline
                "
          
              >
                {link.name}
              </Link>
            </li>
          ))}
<Link to='/signin '>
          <button
            id="mediumBlue-button"
            className="ml-0 md:ml-9 px-6 rounded-2xl py-1"
            type="button"
          >
            {t('Button.si')}
          </button>
           </Link>
          {isLoggedIn && (
            <button
              id="mediumBlue-button"
              className="ml-0 md:ml-9 px-6 rounded-2xl py-1"
              type="button"
              onClick={handleSignOut}
            >
              Sign Out
            </button>
          )}

        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
