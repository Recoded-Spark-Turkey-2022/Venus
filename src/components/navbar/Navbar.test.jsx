// eslint-disable-next-line no-unused-vars
import * as react from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux'
import { store} from '../../app/store'
import renderer from 'react-test-renderer';
import Navbar from "./Navbar"

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: key => key})
}));

it('renders correctly' , () => {
  const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <Navbar />
      </Provider>
    </Router>
  );
  expect(tree).toMatchSnapshot();
});