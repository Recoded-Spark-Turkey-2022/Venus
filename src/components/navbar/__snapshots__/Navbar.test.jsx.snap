// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="shadow-md w-full fixed z-1000 top-0 left-0"
  data-testid="nav"
>
  <div
    className="md:flex items-center justify-between bg-white py-4 md:px-10 px-7"
  >
    <div
      className="font-bold text-2xl cursor-pointer flex items-center font-[Poppins]  text-cyan-600"
    >
      <span
        className="text-3xl text-cyan-600 mr-1 pt-2"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            data-testid="logo-img"
            src="Logo.svg"
          />
        </a>
      </span>
    </div>
    <button
      className="text-3xl absolute right-8 top-6 cursor-pointer md:hidden"
      onClick={[Function]}
      type="button"
    >
      <ion-icon
        name="close"
      >
         
      </ion-icon>
    </button>
    <ul
      className="md:flex md:items-center md:pb-0 pb-12 absolute md:static bg-white md:z-auto z-[-1] left-0 w-full md:w-auto md:pl-0 pl-9 transition-all duration-500 ease-in top-30 "
    >
      <li
        className="md:ml-8 text-xl md:my-0 my-7"
      >
        <a
          className="text-cyan-600 hover:text-cyan-600 duration-500 hover:underline "
          href="/"
          onClick={[Function]}
        >
          Nav.home
        </a>
      </li>
      <li
        className="md:ml-8 text-xl md:my-0 my-7"
      >
        <a
          className="text-cyan-600 hover:text-cyan-600 duration-500 hover:underline "
          href="/about"
          onClick={[Function]}
        >
          Nav.about
        </a>
      </li>
      <li
        className="md:ml-8 text-xl md:my-0 my-7"
      >
        <a
          className="text-cyan-600 hover:text-cyan-600 duration-500 hover:underline "
          href="/blogs"
          onClick={[Function]}
        >
          Nav.blogs
        </a>
      </li>
      <li
        className="md:ml-8 text-xl md:my-0 my-7"
      >
        <a
          className="text-cyan-600 hover:text-cyan-600 duration-500 hover:underline "
          href="/contact"
          onClick={[Function]}
        >
          Nav.contact
        </a>
      </li>
      <a
        href="/signin "
        onClick={[Function]}
      >
        <button
          className="ml-0 md:ml-9 px-6 rounded-2xl py-1"
          id="mediumBlue-button"
          type="button"
        >
          Button.si
        </button>
      </a>
    </ul>
  </div>
</nav>
`;
